[FEAT] Category 추가 제목 입력 뷰 레이아웃 구현

noah0316 opened the pull request:
## 🟣 관련 이슈
- closed: #96 
## 🟣 구현/변경 사항 및 이유
Category 추가 flow에서 카테고리 제목 입력 뷰 레이아웃을 구현하였습니다.
<!-- 구현/변경한 내용과 그 이유를 적어주세요. -->
## 🟣 PR Point
- [x] Rx를 사용해 `titleTextField`에  text가 있는지 확인하도록 구현하였습니다.
- [x] iOS 15부터 사용가능한 UIButton의 `ConfigurationUpdateHandler`를 사용해 버튼의 상태에 따라 UI를 업데이트 하도록 구현했습니다.
<!-- 리뷰어 분들이 집중적으로 보셨으면 하는 내용을 적어주세요 -->

## 🟣 참고 사항
iOS 15부터 사용가능한 `keyboardLayoutGuide`으로 다음 버튼의 bottom layout을 잡으려고 했으나, 
Share Extension에서 적용이 잘 안되는 이슈가 있어 NotificationCenter를 이용하는 방식으로 구현했습니다!
~~IQKeyboardManager 역시 App Extension에서는 사용할 수 없었습니다!~~
<!-- 참고할 사항(+스크린샷)이 있다면 적어주세요. -->
![Simulator Screen Recording - iPhone 13 mini - 2022-01-19 at 15 09 39](https://user-images.githubusercontent.com/63908856/150081503-e320cf6a-cc09-4550-b5e0-769cb020daeb.gif)

noah0316 committed:
[MOVE] ShareExtension 폴더구조 변경
- 카테고리 관련 View는 Category 폴더에서 관리

noah0316 committed:
[ADD] Share Extension SwiftLint 적용(#96)

noah0316 committed:
[FEAT] Category 추가 제목 입력 뷰 레이아웃 구현 (#96)

noah0316 committed:
[CHORE] 프로젝트 파일 추가파일 반영 (#96)

Suyeon9911 reviewed (approved):
고생꽈리 !~~ 수고꽈리 ~~~ 👍🏻

YoonAh-dev commented on Havit/ShareExtension/Screens/Category/View/AddCategoryTitleViewController.swift:
```diff
@@ -0,0 +1,193 @@
+    
+    weak var coordinator: ShareExtensionMainCoordinator?
+    
+    var keyboardHeight: CGFloat = 0
```
p1;

혹시 외부에서 keyboardHeight를 사용하시나요?

noah0316 replied:
아닙니다! 바로 수정해서 커밋하도록 하겠습니다!

noah0316 replied:
반영했습니다!
> 반영 commit : [링크](https://github.com/TeamHavit/Havit-iOS/pull/116/commits/f2bdc650dbe43624c81f04e76cc0ca8acc9be3b5#diff-bfb7a4edad5a1df2fd61c409c0586d5fedd6731d95136785db0f7d26aea5bc58R19)

noah0316 committed:
[CHORE] 코드리뷰 반영 (#96)
- 멤버변수 접근 제한자 변경

YoonAh-dev reviewed (approved):
얼른 뷰를 짜도록 하세요. 화이팅!💜

sujinnaljin commented on Havit/ShareExtension/Screens/Category/View/AddCategoryTitleViewController.swift:
```diff
@@ -0,0 +1,193 @@
+    
+    // MARK: - property
+    
+    weak var coordinator: ShareExtensionMainCoordinator?
```
제가 지웠어여 재삼다

noah0316 replied:
옙 확인했습니다!!!!

sujinnaljin commented on Havit/ShareExtension/Screens/Category/View/AddCategoryTitleViewController.swift:
```diff
@@ -0,0 +1,193 @@
+    // MARK: - @objc
+    
+    @objc
+    private func keyboardWillShow(_ notification: NSNotification) {
```
결국,,

noah0316 replied:
예..그렇게 됐습니다...

sujinnaljin commented on Havit/ShareExtension/Screens/Category/View/AddCategoryTitleViewController.swift:
```diff
@@ -0,0 +1,193 @@
+    private func bind() {
+        titleTextField.rx.text
+            .bind { [weak self] _ in
+                if let isTitleFieldHasText = self?.titleTextField.hasText {
```
p5;
저 뭐냐 
.map{self?.titleTextField.hasText} 
.bind(self?.nextButton.isEnabled)
이런식으로 쓸수도 있을 수 있을거 같습니다??? 지금 거진 손코딩이라 컴파일러는 안될듯!

noah0316 replied:
@sujinnaljin 
말씀 해주신방법이 더 깔끔한 것 같아 반영했습니다!!
> 반영 commit : [링크](https://github.com/TeamHavit/Havit-iOS/pull/116/commits/5e9ecd08b51f86d61d431f82433a88efe999601b#diff-bfb7a4edad5a1df2fd61c409c0586d5fedd6731d95136785db0f7d26aea5bc58R161-R165)

sujinnaljin commented on Havit/ShareExtension/Screens/Category/View/AddCategoryTitleViewController.swift:
```diff
@@ -0,0 +1,193 @@
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        setUpKeyboardForButtonAnimation()
```
p5;
이건 왜 viewWillAppear 에서 설정해줘야해여?

noah0316 replied:
@sujinnaljin 
키보드 올라오는 시점을 확인하다 `viewWillAppear`에서 작동하는 것을 확인해 이곳에서 설정해줬었는데,
`viewDidLoad`에서 설정해주는 것으로 변경했습니다!!
> 반영 commit : [링크](https://github.com/TeamHavit/Havit-iOS/pull/116/commits/5e9ecd08b51f86d61d431f82433a88efe999601b#diff-bfb7a4edad5a1df2fd61c409c0586d5fedd6731d95136785db0f7d26aea5bc58L111-L114)

noah0316 replied:
@sujinnaljin 
키보드 올라오는 시점을 확인하다 `viewWillAppear`에서 작동하는 것을 확인해 이곳에서 설정해줬었는데,
`viewDidLoad`에서 설정해주는 것으로 변경했습니다!!
> 반영 commit : [링크](https://github.com/TeamHavit/Havit-iOS/pull/116/commits/5e9ecd08b51f86d61d431f82433a88efe999601b#diff-bfb7a4edad5a1df2fd61c409c0586d5fedd6731d95136785db0f7d26aea5bc58L111-L114)

sujinnaljin reviewed (approved):
수고꽐~

noah0316 committed:
[CHORE] 코드리뷰 반영 (#96)
- coordinator 멤버변수 제거
- setUpKeyboardForButtonAnimation 메소드 호출 시점 변경
- titleTextField text확인 로직 변경

noah0316 committed:
Merge branch 'develop' of https://github.com/TeamHavit/Havit-iOS into feature/96-category-title
